chapters/chap03-sec02.tex

```diff
@@ -18,28 +18,33 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
 
 \subsection{\mimc{}-based compression function}\label{instantiation:mkhash:mimc-compressionf}
 
-There exists two main techniques to construct a hash function from a block-cipher (or permutation): sponge functions~\cite{bertoni2007sponge} and iterated compression functions~\cite{black2002black}.
-In a Merkle tree, we need to hash two fixed size inputs into one. This is the definition of a compression function. As such, we decided to use a compression function and settled for Miyaguchi-Preneel compression function~\cite[$f_3$ function]{black2002black} construct for its proven security (it is more secure than the more common Davies-Meyer construct~\cite[$f_5$ function]{black2002black}, and we do not require the flexibility of the latter~\cite[Section 3]{gazzoni2006maelstrom}).
+There exist two main techniques to construct a hash function from a block-cipher (or permutation): sponge functions~\cite{bertoni2007sponge} and iterated compression functions~\cite{black2002black}.
+
+A Merkle tree consists of values of a fixed size, where the values in each ``layer'' are generated by hasing pairs of values from the previous ``layer''. That is, we require a compression function $\mkhash : \FFx{\rCURVE} \times \FFx{\rCURVE} \to \FFx{\rCURVE}$. (Miyaguchi-Preneel is more secure~\cite[$f_5$ function]{black2002black} than the more flexible Davies-Meyer construct~\cite[Section 3]{gazzoni2006maelstrom}, but this flexibility is not required in our case).
 
 \subsubsection{Miyaguchi-Preneel compression construct}
 
-Miyaguchi-Preneel (MP)~\cite[$f_3$ function]{black2002black} is a general construction that allows to build compression functions from block ciphers (see~\cref{preliminaries:definitions:hashcomp}). Given a block cipher \Enc, we denote the corresponding compression function by \fMP{}, its description is given in~\cref{instantiation:fig:mp-constructions}. We focus on ciphers working on \FFx{p}, for some prime $p$, while the original construction is defined over binary fields. As such, we replace the bitwise addition $\oplus$ by the modular addition in \FFx{\rBN} (see~\cite{mp-security-ethsnarks}).
+Miyaguchi-Preneel (MP)~\cite[$f_3$ function]{black2002black} is a general scheme for constructing compression functions from block ciphers (see~\cref{preliminaries:definitions:hashcomp}). Given a block cipher \Enc, the corresponding compression function by \fMP{} is given in~\cref{instantiation:fig:mp-constructions}. The original construction is defined over binary fields, however \zeth~operates over the field $\FFx{\rCURVE}$. Hence, the general discussion here replaces the bitwise addition operator $\oplus$ by modular addition in \FFx{p} for some prime $p$ (see~\cite{mp-security-ethsnarks}).
```

Comment:
Switched to `\zeth~operates over a prime field.`  (If we keep a separate unbound $p$ in this chapter, it seems incorrect to say that `\zeth~operates over the field $\FFx{p}$.`, since we use a different variable $\rCURVE$ in the description.)